### PR TITLE
Xrun support

### PR DESCRIPTION
**Pull request by AndrewNolte**
Run tests with --xrun

ex: test_regress/t/t_timing_delay_callstack.pl --xrun

### CODE REVIEW

File: test_regress/driver.pl [hunk 1]
```diff
@@ -51,6 +51,7 @@ BEGIN
        iv    => [          "simulator", "simulator_st", "iv"],
        ms    => ["linter", "simulator", "simulator_st", "ms"],
        nc    => ["linter", "simulator", "simulator_st", "nc"],
+       xrun  => ["linter", "simulator", "simulator_st", "xrun"],
        vcs   => ["linter", "simulator", "simulator_st", "vcs"],
        xsim  => ["linter", "simulator", "simulator_st", "xsim"],
```
Comment (wsnyder) [inline test_regress/driver.pl:56]: Please put into sorted location just above xsim.  (And if wish cleanup other missorts).

```suggestion
       vcs   => ["linter", "simulator", "simulator_st", "vcs"],
       xrun  => ["linter", "simulator", "simulator_st", "xrun"],
       xsim  => ["linter", "simulator", "simulator_st", "xsim"],
```

File: test_regress/driver.pl [hunk 2]
```diff
@@ -122,6 +123,7 @@ BEGIN
           "iv!"         => sub { $opt_scenarios{iv} = $_[1]; },
           "ms!"         => sub { $opt_scenarios{ms} = $_[1]; },
           "nc!"         => sub { $opt_scenarios{nc} = $_[1]; },
+          "xrun!"       => sub { $opt_scenarios{xrun} = $_[1]; },
           "vlt!"        => sub { $opt_scenarios{vlt} = $_[1]; },
           "vltmt!"      => sub { $opt_scenarios{vltmt} = $_[1]; },
           "vcs!"        => sub { $opt_scenarios{vcs} = $_[1]; },
```
Comment (wsnyder) [inline test_regress/driver.pl:129]:
```suggestion
          "vcs!"        => sub { $opt_scenarios{vcs} = $_[1]; },
          "vlt!"        => sub { $opt_scenarios{vlt} = $_[1]; },
          "vltmt!"      => sub { $opt_scenarios{vltmt} = $_[1]; },
          "xrun!"       => sub { $opt_scenarios{xrun} = $_[1]; },
```

File: test_regress/driver.pl [hunk 3]
```diff
@@ -587,6 +589,7 @@ sub new {
     $self->{scenario} ||= "vlt" if $self->{vlt};
     $self->{scenario} ||= "vltmt" if $self->{vltmt};
     $self->{scenario} ||= "nc" if $self->{nc};
+    $self->{scenario} ||= "xrun" if $self->{xrun};
     $self->{scenario} ||= "ms" if $self->{ms};
     $self->{scenario} ||= "iv" if $self->{iv};
     $self->{scenario} ||= "xsim" if $self->{xsim};
```
Comment (wsnyder) [inline test_regress/driver.pl:595]:
```suggestion
    $self->{scenario} ||= "iv" if $self->{iv};
    $self->{scenario} ||= "ms" if $self->{ms};
    $self->{scenario} ||= "xrun" if $self->{xrun};
    $self->{scenario} ||= "xsim" if $self->{xsim};
```

**Review by wsnyder — COMMENTED** [review]
Thanks, also please update internals.rst to document --xrun.

**Comment by AndrewNolte** [comment]
I changed them to be before xsim, and followed the order in the docs. I think it might be worth just refactoring driver.pl into a declarative / class structure, and maybe changing it to python

**Comment by wsnyder** [comment]
Converting to python is a project I started but ended up very large, and the different versions of python people have can also cause problems.  Perl maybe old now but is at least stable.

**Pull request merged**